python/cudf/dataframe/columnops.py

```diff
@@ -143,12 +145,12 @@ def column_select_by_position(column, positions):
     Returns (selected_column, selected_positions)
     """
     from cudf.dataframe.numerical import NumericalColumn
-    assert column.null_count == 0
 
-    selvals = cudautils.gather(column.data.to_gpu_array(),
-                               positions.data.to_gpu_array())
+    assert column.null_count == 0
 
-    selected_values = column.replace(data=Buffer(selvals))
+    selvals = cpp_copying.apply_gather_array(column.data.to_gpu_array(),
+                                             positions.data.to_gpu_array())
+    selected_values = column.replace(data=selvals.data)
     selected_index = Buffer(positions.data.to_gpu_array())
 
     return selected_values, NumericalColumn(data=selected_index,
```

Comment:
Remove the null count assertion, replace this with a column gather instead of an array gather.